src/LanguageServer.ts

```diff
@@ -1260,7 +1261,7 @@ export class LanguageServer {
 
     @AddStackToErrorMessage
     @TrackBusyStatus
-    private async onFullSemanticTokens(params: SemanticTokensParams) {
+    private async onFullSemanticTokens(params: SemanticTokensParams): Promise<SemanticTokens | undefined> {
```

Comment:
This could always return `undefined`; the signature change is just to make it explicit.